server/backend/backend.go

```diff
@@ -44,15 +45,16 @@ type Backend struct {
 
 func New(authLife time.Duration, domain string) *Backend {
```

Comment:
it might be simpler to take in the filter as a parameter? Then you don't need `SetSystemLabelFilter` method.

- create server (either with default filter, or optionally with a build option to set a custom filter)
- server creates backend, passing in the filter (either default or custom)
- no need to have methods "SetSystemLabelFilter" on server/backend at all since there will always be a filter configured via the builder